tests/testthat/test-plotly-subplot.R

```diff
@@ -208,6 +208,56 @@ test_that("shape paper repositioning", {
   expect_equal(y1, rep(0.75, 2))
   expect_equal(xref, rep("paper", 2))
   expect_equal(yref, rep("paper", 2))
+  
+  # now with a fixed height/width
+  p1 <- plot_ly(mtcars) %>%
+    layout(
+      shapes = ~list(
+        type = "rect",
+        x0 = 0.25,
+        x1 = 0.75,
+        xref = "paper",
+        y0 = 0,
+        y1 = 30,
+        yanchor = 0.5,
+        ysizemode = "pixel",
+        yref = "paper",
+        fillcolor = "red"
+      )
+    )
+  p2 <- plot_ly(mtcars) %>%
+    layout(
+      shapes = ~list(
+        type = "rect",
+        y0 = 0.25,
+        y1 = 0.75,
+        yref = "paper",
+        x0 = 0,
+        x1 = 30,
+        xanchor = 0.5,
+        xsizemode = "pixel",
+        xref = "paper",
+        line = list(color = "blue")
+      )
+    )
+  
+  s <- subplot(p1, p2)
+  shapes <- expect_doppelganger_built(s, "subplot-reposition-shape-fixed")$layout$shapes
```

Comment:
Add a rect shape after-the-fact as well

Reply:
I forgot that you can't currently add shapes to a plot that already has shapes -- https://github.com/ropensci/plotly/issues/1333